### PR TITLE
Add transform to handle empty box as training data

### LINKED ISSUE #5990: ConvertBoxToStandardModed cannot handle empty boxes
**Describe the bug**
When input boxes are empty [], it got error since the boxes are supposed to have shape [N,4] or [N,6].

ValueError: Currently we support only boxes with shape [N,4] or [N,6], got boxes with shape torch.Size([1, 0])

**To Reproduce**
2. Install '....'
3. Run commands '....'

**Expected behavior**

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes #5990 .

### Description

Add transforms to convert empty box with shape (0,M) or (0,) into (0,4) or (0,6).
Provide format checking inside detector so users can know how to solve the format issue with empty box input.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/detection/transforms/dictionary.py [hunk 1]
```diff
@@ -95,6 +99,50 @@
 DEFAULT_POST_FIX = PostFix.meta()
 
 
+class StandardizeEmptyBoxd(MapTransform, InvertibleTransform):
```
Comment (wyli) [inline monai/apps/detection/transforms/dictionary.py:102]: I'm not familiar with the use cases, is there a case that the user don't want to use this transform... maybe this transform should be enabled by default within some existing transforms, for example in  `MaskToBoxD`?

Comment (Can-Zhao) [reply monai/apps/detection/transforms/dictionary.py:102]: Good point. How about running it by default inside the detector?

Comment (wyli) [reply monai/apps/detection/transforms/dictionary.py:102]: sure, perhaps call the util in the detector https://github.com/Can-Zhao/MONAI/blob/c13895d021b0d5aa4caccc85f742a7dac4b1cc89/monai/data/box_utils.py#L519

Comment (Can-Zhao) [reply monai/apps/detection/transforms/dictionary.py:102]: I let the detector automatically correct the empty box shape and label data type, while print a warning message to let the users know what has been done. Shall we remove the array and dict transforms?

**Review by wyli — APPROVED** [review]

File: monai/apps/detection/transforms/dictionary.py (hunk 1 above)
Comment (wyli) [reply monai/apps/detection/transforms/dictionary.py:102]: looks good to me!

**Pull request merged**